workspaces/ballerina/ballerina-extension/src/utils/auth-utils.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+
+/**
+ * Checks if we're running in a cloud editor environment
+ * @returns boolean - true if CLOUD_STS_TOKEN environment variable exists
+ */
+export function isCloudEditorEnvironment(): boolean {
+    return !!process.env.CLOUD_STS_TOKEN;
+}
+
+/**
+ * Reads the current access token from Settings.toml
+ * @returns Promise<string | undefined> - The current access token or undefined if not found
+ */
+export async function getCurrentAccessToken(): Promise<string | undefined> {
+    try {
+        const settingsPath = path.join(os.homedir(), '.ballerina', 'Settings.toml');
```

Comment:
we can consider using `extension.ballerinaExtInstance.getBallerinaUserHome()` to get the `.ballerina` directory

Reply:
Addressed with https://github.com/wso2/vscode-extensions/pull/369/commits/d7593a26e6842f454a4eb6e9460eeb3bb3bac570